File: leetcode/all/python3/219.存在重复元素-ii.py
#
# @lc app=leetcode.cn id=219 lang=python3
#
# [219] 存在重复元素 II
#

# @lc code=start
from typing import List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        st = set()
        for i, n in enumerate(nums):
            if i > k:
                st.remove(nums[i - k - 1])
            if n in st:
                return True
            st.add(n)
        return False


# @lc code=end

File: leetcode/all/typescript/219.存在重复元素-ii.ts
/*
 * @lc app=leetcode.cn id=219 lang=typescript
 *
 * [219] 存在重复元素 II
 */

// @lc code=start
function containsNearbyDuplicate(nums: number[], k: number): boolean {
  const set = new Set<number>()
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i]
    if (i > k) set.delete(nums[i - k - 1])
    if (set.has(num)) return true
    set.add(num)
  }
  return false
}
// @lc code=end
